connector/params.go

```diff
@@ -1,63 +1,87 @@
+// nolint
```

Comment:
The `connector` namespace felt like a good place to have this, so moved the current auth/generic connectors out into `providers/generic`. This also gets around import cycle issues (this has minimum disruption)